### PR TITLE
Add find command by all prefixes

### LINKED ISSUE #93: Add partial name search

### PR DESCRIPTION
**Pull request by haleemairfan**
Fixes #93

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/prefix/PrefixHandler.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * @param lastShownList The list of persons to search through.
+     * @param attribute The attribute to match against (e.g., a phone number, email, or address).
+     * @return A list of persons whose attributes match the provided attribute.
+     */
+    public List<Person> findPersonByAttribute(List<Person> lastShownList, Object attribute) {
+        String type = getTypeOfAttribute(attribute);
+        List<Person> matchingPersons = new ArrayList<>();
+        for (Person person : lastShownList) {
+            switch (type) {
+            case "NAME":
```
Comment (miuna9890) [inline src/main/java/seedu/address/logic/parser/prefix/PrefixHandler.java:99]: for the finding name feature it only finds by the first name. for list 1. Mithuna hi 2. Anna bye, [find n/hi] returns 0 persons listed but i believe from our UG it should return Mithuna hi. also [find n/ anna hi] should return both mithuna hi and anna bye rite cause we allow partial name searches?

**Review by miuna9890 — COMMENTED** [review]
the other find attributes seems to work well just the finding by name command.

**Comment by zhou-colla** [comment]
Should we allow partial matching? Find function only works for full match, partial match does not work.

Command: 

image

Exception:
image

**Review by yyueyii — APPROVED** [review]
Works well for me!

**Pull request merged**